crates/bevy_reflect/src/serde/de.rs

```diff
@@ -390,221 +495,449 @@ impl<'a, 'de> Visitor<'de> for MapVisitor<'a> {
     }
 }
 
-struct StructDeserializer<'a> {
-    registry: &'a TypeRegistry,
-}
-
-impl<'a, 'de> DeserializeSeed<'de> for StructDeserializer<'a> {
-    type Value = DynamicStruct;
-
-    fn deserialize<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        deserializer.deserialize_map(StructVisitor {
-            registry: self.registry,
-        })
-    }
-}
-
-struct StructVisitor<'a> {
+struct EnumVisitor<'a> {
+    enum_info: &'a EnumInfo,
     registry: &'a TypeRegistry,
 }
 
-impl<'a, 'de> Visitor<'de> for StructVisitor<'a> {
-    type Value = DynamicStruct;
+impl<'a, 'de> Visitor<'de> for EnumVisitor<'a> {
+    type Value = DynamicEnum;
 
     fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
-        formatter.write_str("struct value")
+        formatter.write_str("reflected enum value")
     }
 
     fn visit_map<V>(self, mut map: V) -> Result<Self::Value, V::Error>
     where
         V: MapAccess<'de>,
     {
-        let mut dynamic_struct = DynamicStruct::default();
-        while let Some(key) = map.next_key::<String>()? {
-            let value = map.next_value_seed(ReflectDeserializer {
-                registry: self.registry,
-            })?;
-            dynamic_struct.insert_boxed(&key, value);
+        let variant_name = map
+            .next_key::<&str>()?
```

Comment:
I haven't encountered the same problem as in my previous comment, however, this is also a potential point of failure.